csrc/multidevice/utils.cpp

```diff
@@ -367,6 +368,15 @@ bool haveDifferentShardings(
     return true;
   }
 
+  // Special handling of SelectOp for a quick fix
+  // TODO: work on a proper implementation
+  if (consumer->definition()->isA<SelectOp>()) {
+    auto* select_op = consumer->definition()->as<SelectOp>();
+    NVF_ERROR(
+        select_op->input(0) == producer, "SelectOp input 0 is not producer");
+    return select_op->getIndexedID()->isDeviceDim();
```

Comment:
Thanks for extracting this into a separate PR! 

Can you code-comment the logic here? I don't follow this at the first glance.

Reply:
If you select into the sharded axis, then select Op is resharding because the consumer doesn't contain this axis therefore this axis becomes "replicated" on `my_device`. Does it make sense or did you have another model in mind?
(I am realizing that we should also analyze whether the other axis's sharding remain unchanged)


The long term better solution would actually to "select" into the DeviceMesh, e.g.,
```cpp
tv0 = makeContigTensor(2); // [DIDx(4), 8] on mesh {0,1,2,3}
tv1 = select(tv0, /*axis=*/0, /*index=*/1); // [8] on mesh {1}
```
But for achieving this with symbolic "index" we need to make DeviceMesh symbolic.

Reply:
> (I am realizing that we should also analyze whether the other axis's sharding remain unchanged)

I've added that in the last commit